### PR TITLE
test(trtllm): speed up TRTLLM gpu test time by reducing imports

### PR DESCRIPTION
**Pull request by rmccorm4**
## Summary

Reduce TRTLLM test job time from ~25min to ~19min.

- assign independent system-status ports to the TensorRT-LLM multimodal prefill and decode workers
- gate `pd_multimodal` startup on active one-token canary health checks instead of a fixed 120-second delay
- move 79 import-heavy, zero-VRAM TensorRT-LLM tests out of the per-node GPU-parallel orchestrator and into the existing single-process GPU stage to amortize a single TRTLLM import across all of the fast tests instead of paying the slow import cost across all parallel processes.
- retain the dedicated model-predownload sentinel and the real 2 GiB CUDA logits test in the parallel pool

## Root cause

`pd_multimodal` used a fixed startup sleep because the prefill and decode workers shared a system-status port. A readiness check against that shared port could report the wrong worker as ready. Independent ports make per-worker active readiness probes reliable.

Separately, the GPU-parallel orchestrator launches one pytest subprocess per selected test node. In [run 30510438808](https://github.com/ai-dynamo/dynamo/actions/runs/30510438808/job/90773002784?pr=12413), 79 import-heavy tests were marked `profiled_vram_gib(0)`. They used no GPU memory but each paid pytest and TensorRT-LLM import/setup overhead, consuming process slots and delaying queued positive-VRAM workloads while the GPU was nearly idle.

The rc22 request-ID compatibility behavior is unchanged by this patch.

## Impact

The multimodal test now advances as soon as both workers successfully serve a canary request, while retaining bounded timeouts and useful failure logs.

The moved tests remain marked `pre_merge`, `trtllm`, and `gpu_1`. Omitting only `profiled_vram_gib(0)` routes them through the existing sequential GPU expression, preserving their CUDA-visible runtime while sharing one TensorRT-LLM initialization. Baseline logs show the 79 nodes accounted for about 4,087 child-seconds and delayed the next 20 GiB workload by about 372 seconds. The expected wall-clock reduction is roughly six minutes, pending a new full-CI run.

- Linear: [DIS-2552](https://linear.app/nvidia/issue/DIS-2552)
- Original failure: [run 30504008796](https://github.com/ai-dynamo/dynamo/actions/runs/30504008796/attempts/1?pr=12182)
- Runtime baseline: [run 30510438808](https://github.com/ai-dynamo/dynamo/actions/runs/30510438808/job/90773002784?pr=12413)

## Validation

- `pre-commit` passed for all eight marker-routing files
- `bash -n tests/backends/trtllm/common.sh tests/backends/trtllm/agg_multimodal.sh tests/backends/trtllm/pd_multimodal.sh`
- focused multimodal helper tests: 7 passed, 5 skipped
- marker audit confirms the real CUDA logits test retains `profiled_vram_gib(2.0)` and the predownload sentinel retains `profiled_vram_gib(0)`
- full TensorRT-LLM GPU E2E validation is pending CI; the local environment does not include `tensorrt_llm`

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented system-status port collisions between disaggregated multimodal prefill and decode workers.
  * Improved worker readiness checks by requiring a successful health canary before reporting readiness.

* **Tests**
  * Updated GPU test scheduling and documentation to support shared TensorRT-LLM initialization and avoid unnecessary VRAM profiling.

### CODE REVIEW

File: tests/serve/test_trtllm.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ class TRTLLMConfig(EngineConfig):
         model="Qwen/Qwen3-VL-2B-Instruct",
         frontend_port=DefaultPort.FRONTEND.value,
         timeout=900,
-        delayed_start=120,
+        health_check_workers=True,
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply tests/serve/test_trtllm.py:360]: Addressed by [`a14d4f7f8f`](https://github.com/ai-dynamo/dynamo/commit/a14d4f7f8ff77d216b4fe446691188d1e15a5f29), which added an explicit pytest timeout marker. [`ed24f2ca64`](https://github.com/ai-dynamo/dynamo/commit/ed24f2ca64a17121729be78e70aec32ab73903d1) subsequently set it to 360 seconds. On exact head [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c), `pd_multimodal` passed in 106s/360s in [full CI](https://github.com/ai-dynamo/dynamo/actions/runs/30669844224/job/91288621673).

File: tests/serve/test_trtllm.py [hunk 2]
```diff
@@ -366,6 +367,9 @@ class TRTLLMConfig(EngineConfig):
         env={
             "PREFILL_CUDA_VISIBLE_DEVICES": "0",
             "DECODE_CUDA_VISIBLE_DEVICES": "0",
+            # Make worker /health readiness depend on a successful one-token
+            # engine canary instead of the system-status server alone.
+            "DYN_HEALTH_CHECK_ENABLED": "true",
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply tests/serve/test_trtllm.py:372]: Resolved by exact-head validation at [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c). With the worker health gates enabled, `pd_multimodal` passed in 106s/360s in [the TRT-LLM GPU job](https://github.com/ai-dynamo/dynamo/actions/runs/30669844224/job/91288621673), confirming the multimodal disaggregated canary path succeeds.

File: tests/serve/test_trtllm.py [hunk 3]
```diff
@@ -352,11 +352,12 @@ class TRTLLMConfig(EngineConfig):
             pytest.mark.pre_merge,
             pytest.mark.profiled_vram_gib(15.0),
             pytest.mark.requested_trtllm_kv_tokens(1056),
+            pytest.mark.timeout(360),  # 3x measured 118s CI runtime
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply tests/serve/test_trtllm.py:355]: Addressed by [`ed24f2ca64`](https://github.com/ai-dynamo/dynamo/commit/ed24f2ca64a17121729be78e70aec32ab73903d1) (360s pytest timeout) and [`1403023f8d`](https://github.com/ai-dynamo/dynamo/commit/1403023f8d91f5d9a7d0c8247c43870ac5a70a80) (300s readiness budget), then validated on exact head [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c). `pd_multimodal` passed in 105s and 106s in two full GPU runs; the 300s harness deadline remains below the 360s outer timeout so harness diagnostics run first.

File: examples/backends/trtllm/launch/disagg_multimodal.sh [hunk 4]
```diff
@@ -28,11 +28,18 @@ fi
 HTTP_PORT="${DYN_HTTP_PORT:-8000}"
 print_launch_banner --multimodal "Launching Disaggregated Multimodal Serving (2 GPUs)" "$MODEL_PATH" "$HTTP_PORT"
 
+# Set worker ports before launching the frontend so it does not inherit the shared port.
+SYSTEM_PORT_BASE="${DYN_SYSTEM_PORT:-8081}"
+DYN_SYSTEM_PORT1="${DYN_SYSTEM_PORT1:-$SYSTEM_PORT_BASE}"
+DYN_SYSTEM_PORT2="${DYN_SYSTEM_PORT2:-$((SYSTEM_PORT_BASE + 1))}"
+unset DYN_SYSTEM_PORT
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply examples/backends/trtllm/launch/disagg_multimodal.sh]: Fixed by [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c). The `base + 1` derivation was removed; the launcher now passes the independently allocated `DYN_SYSTEM_PORT1` and `DYN_SYSTEM_PORT2` values directly to the two workers.

File: tests/serve/test_trtllm.py [hunk 5]
```diff
@@ -352,11 +352,12 @@ class TRTLLMConfig(EngineConfig):
...
         ],
         model="Qwen/Qwen3-VL-2B-Instruct",
         frontend_port=DefaultPort.FRONTEND.value,
-        timeout=900,
-        delayed_start=120,
+        timeout=300,
+        health_check_workers=True,
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply tests/serve/test_trtllm.py:360]: Addressed by [`ed24f2ca64`](https://github.com/ai-dynamo/dynamo/commit/ed24f2ca64a17121729be78e70aec32ab73903d1) and [`1403023f8d`](https://github.com/ai-dynamo/dynamo/commit/1403023f8d91f5d9a7d0c8247c43870ac5a70a80), which set the outer timeout to 360s and the readiness budget to 300s. Exact head [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c) then passed `pd_multimodal` in 106s/360s in [full CI](https://github.com/ai-dynamo/dynamo/actions/runs/30669844224/job/91288621673).

File: tests/serve/test_trtllm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmccorm4) [reply tests/serve/test_trtllm.py:372]: Resolved by exact-head validation at [`e72898f82f`](https://github.com/ai-dynamo/dynamo/commit/e72898f82fdc40583abf99b0108b7eaa82c5fc0c). `pd_multimodal` passed in 106s/360s in [the TRT-LLM GPU job](https://github.com/ai-dynamo/dynamo/actions/runs/30669844224/job/91288621673), confirming the canary fallback works with the health-gated workers.

**Review by KrishnanPrash — APPROVED** [review]

**Review by ziqifan617 — APPROVED** [review]
